full-service/src/json_rpc/wallet_api.rs

```diff
@@ -0,0 +1,877 @@
+// Copyright (c) 2020-2021 MobileCoin Inc.
+
+//! Wallet API.
+//!
+//! A [JSON RPC 2.0](https://www.jsonrpc.org/specification) API inspired by [Stratum](https://github.com/aeternity/protocol/blob/master/STRATUM.md) .
+
+use crate::db::AccountID;
+use crate::json_rpc::{self, Request, Response};
+use crate::service::Wallet;
+use rocket::{get, post, routes};
+use rocket_contrib::json::Json;
+use serde_json::Map;
+use std::iter::FromIterator;
+use std::sync::{Arc, Mutex};
+use strum::IntoEnumIterator;
+
+/// Managed state for the Wallet API. This object must be thread-safe.
+pub struct WalletApiState {
+    pub service: Arc<Mutex<dyn Wallet + Send + Sync + 'static>>,
+}
+
+// The Wallet API inner method, which handles switching on the method enum.
+//
+// Note that this is structured this way so that the routes can be defined to take
+// explicit Rocket state, and then pass the service to the inner method. This allows
+// us to properly construct state with Mock Connection Objects in tests.
+fn wallet_api_inner(
+    service: Arc<Mutex<dyn Wallet>>,
+    command: Json<Request>,
+) -> Result<Json<Response>, String> {
+    let service = service.lock().unwrap();
+
+    let result = match command.0 {
+        Request::create_account { name, first_block } => {
+            let fb = first_block
+                .map(|fb| fb.parse::<u64>())
+                .transpose()
+                .map_err(|e| format!("{{\"error\": \"{:?}\"}}", e))?;
```

Comment:
I am pretty sure the project now has a better way of doing this... @sugargoat do you remember where you fixed that?

Reply:
It's in the encrypted DB PR, I will be pulling it out as a separate PR - that can happen either before or after this goes in.